Document lobbies (#337)
discord-userdoccers/discord-userdoccers: pages/topics/gateway-events.mdx

}
```

#### Lobby Voice States

> Sent by the client to update user's voice states in multiple lobbies. Fires a [Lobby Voice State Update](#lobby-voice-state-update) and optionally [Lobby Voice Server Update](#lobby-voice-server-update) Gateway event.

Reviewer: ```suggestion
Sent by the client to update the user's voice state in multiple lobbies. Fires a [Lobby Voice State Update](#lobby-voice-state-update) and optionally [Lobby Voice Server Update](#lobby-voice-server-update) Gateway event.
```

> The inner `d` key is array of following objects:

Reviewer: ```suggestion
The inner `d` key is an array of the following objects:
```

###### Update Lobby Voice State Structure

| Field              | Type          | Description                                                                         |
| ------------------ | ------------- | ----------------------------------------------------------------------------------- |
| lobby_id           | snowflake     | The ID of the lobby                                                                 |
| self_mute          | boolean       | Whether the client is muted                                                         |
| self_deaf          | boolean       | Whether the client is deafened                                                      |
| self_video?        | boolean       | Whether the client is streaming video to the channel                                |
| preferred_region?  | string        | The preferred [voice region](/resources/voice#voice-region-object) ID for the lobby |
| preferred_regions? | array[string] | The ranked [voice region](/resources/voice#voice-region-object) IDs for the lobby   |

> {/* TODO: it seems gateway completely ignores preferred_region and preferred_regions, but sent by SDK??? Maybe remove? */}

Reviewer: i say leave it

}
```

> #### Lobby Voice States

Reviewer: ```suggestion
#### Update Lobby Voice States
```
add to list of ops above too

| dm_access_type | integer           | The [DM access level](/resources/relationships#dm-access-type) for the relationship |
| user_id        | snowflake         | The ID of the current user                                                          |

### Lobbies

#### Gateway Lobby Object

[Lobby](/resources/lobby#lobby-object) objects received over the Gateway have extended attributes that are not provided in the REST API.

###### Gateway Lobby Structure

| Field        | Type                                                                   | Description                                                                          |
| ------------ | ---------------------------------------------------------------------- | ------------------------------------------------------------------------------------ |
| voice_states | array[[lobby voice state](/resources/voice#voice-state-object) object] | The voice states of the users already in the lobby                                   |
| region       | string                                                                 | The [voice region](/resources/voice#voice-region-object) ID the lobby is hosted from |

> | metadata     | ?string \| map[string, string]                                         | The metadata of the lobby, might be serialized as a JSON string                      |

Reviewer: pls clarify when this is a string 😭
maybe this field accepts arbitrary data?

###### Lobby Deletion Reason

| Value   | Description           |
| ------- | --------------------- |
| deleted | The lobby was deleted |
| removed | The user was removed  |

#### Lobby Create

Sent when a lobby is updated. The inner payload is a [gateway lobby](#gateway-lobby-object) object.

#### Lobby Update

Sent when a lobby is updated. The inner payload is a [gateway lobby](#gateway-lobby-object) object.

#### Lobby Delete

> Sent when a lobby is deleted, or user was removed from lobby.

Reviewer: ```suggestion
Sent when a lobby is deleted or the user is removed from a lobby.
```

| member         | [lobby member](/resources/lobby#lobby-member-object) object | The updated member                               |
| lobby_id       | snowflake                                                   | The ID of the lobby                              |
| application_id | snowflake                                                   | The ID of the application that created the lobby |

#### Lobby Member Remove

Sent when a lobby member is removed.

###### Lobby Member Remove Structure

| Field          | Type                                                        | Description                                      |
| -------------- | ----------------------------------------------------------- | ------------------------------------------------ |
| member         | [lobby member](/resources/lobby#lobby-member-object) object | The removed member                               |
| lobby_id       | snowflake                                                   | The ID of the lobby                              |
| application_id | snowflake                                                   | The ID of the application that created the lobby |

#### Lobby Message Create

> Sent when a lobby message is created. The inner payload is [message](/resources/message#message-object) object with the [extra structure](#message-object-extra-fields) above.

Reviewer: ```suggestion
Sent when a lobby message is created. The inner payload is a [message](/resources/message#message-object) object with the [extra structure](#message-object-extra-fields) above.
```

<Alert type="warn">

> The message can be a partial [message](/resources/message#partial-message-object) object.

Reviewer: ```suggestion
When the lobby does not have a channel linked, the message will be a partial [message](/resources/message#partial-message-object) object.
```
not sure if this is the actual requirement but whatever it is should be mentioned

</Alert>

#### Lobby Message Update

Sent when a lobby message is updated. The inner payload is a [message](/resources/message#message-object) object with the [extra structure](#message-object-extra-fields) above.

<Alert type="warn">

> The message can be a partial [message](/resources/message#partial-message-object) object.

Reviewer: same note
